src/Compilers/Core/Portable/Compilation/IExpression.cs

```diff
@@ -795,31 +795,21 @@ public interface IObjectCreationExpression : IExpression
         /// <summary>
         /// Explicitly-specified member initializers.
         /// </summary>
-        ImmutableArray<IMemberInitializer> MemberInitializers { get; }
+        ImmutableArray<ISymbolInitializer> MemberInitializers { get; }
     }
 
     /// <summary>
     /// Represents an object member initializer.
     /// </summary>
-    public interface IMemberInitializer : IOperation
+    public interface ISymbolInitializer : IOperation
     {
-        MemberInitializerKind MemberInitializerKind { get; }
         IExpression Value { get; }
     }
-
-    /// <summary>
-    /// Kinds of member initializers.
-    /// </summary>
-    public enum MemberInitializerKind
-    {
-        Field,
-        Property
-    }
-
+    
     /// <summary>
     /// Represents an initialization of a field.
     /// </summary>
-    public interface IFieldInitializer : IMemberInitializer
+    public interface IFieldInitializer : ISymbolInitializer
```

Comment:
@michaelcfanning FYI. You might find it amusing or interesting that the issue they wrestled with here ("there's almost always only one of something, but we _have_ to support the rare case where there can be two or more of them") is the same one we keep encountering in defining the SARIF format. And they came to the same conclusion we did: "Always represent the thing as a collection, even though it will almost always have only one element."